cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -71,3 +74,27 @@ func (r *Reporter) send(spans []*model.Span, process *model.Process) error {
 	}
 	return err
 }
+
+// addTags appends jaeger tags for the agent to every span it sends to the collector.
+func addProcessTags(spans []*model.Span, process *model.Process, agentTags []model.KeyValue) ([]*model.Span, *model.Process) {
+	if process != nil {
+		process.Tags = append(process.Tags, agentTags...)
+		return spans, process
+	}
+	for _, span := range spans {
+		if len(agentTags) > 0 {
+			span.Process.Tags = append(span.Process.Tags, agentTags...)
+		}
+	}
+	return spans, nil
+}
+
+func makeModelKeyValue(agentTags map[string]string) []model.KeyValue {
+	tags := []model.KeyValue{}
```

Comment:
`make([]model.KeyValue, 0, len(agentTags))`

Reply:
This would involve an iterator like so -
```
func makeModelKeyValue(agentTags map[string]string) []model.KeyValue {
	tags := make([]model.KeyValue, 0, len(agentTags))
	i := 0
	for k, v := range agentTags {
		tag := model.String(k, v)
		tags[i] = tag
		i++
	}

	return tags
}
``` 

Should I change to this if it looks better?

Reply:
it doesn't change the loop, because the slice is allocated with 0 initial length, so append() works the same way (but without extra allocations).